lib/queue.rb

```diff
@@ -1,31 +1,75 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @size = 20
+    @queue = []
+    @front = @rear = -1
   end
 
+  # reference: https://learn-2.galvanize.com/cohorts/2173/blocks/867/content_files/04-Stacks-And-Queues/01-stacks-and-queues.md
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    # The Queue is full
+    if ((front == 0 && rear == @size - 1) || (@rear == (@front - 1) % (@size - 1)))    
+      raise ArgumentError.new('The queue is full')
+    # Queue is empty
+    elsif (@front == -1) 
+      @front = 0
+      @rear = 0
+      @queue[@rear] = element
+    # rear needs to wrap around
+    elsif (@rear == @size - 1 && @front != 0) 
+      @rear = 0
+      @queue[@rear] = element
+    else
+      @rear = @rear + 1
+      @queue[@rear] = element
+    end
```

Comment:
This can be simplified a bit.

```suggestion
    elsif (@rear == @size - 1 && @front != 0) 
      @rear = (@rear + 1) % @size
      @queue[@rear] = element
    end
```